sklearn/tree/_utils.pyx

```diff
@@ -62,7 +62,10 @@ cdef inline UINT32_t our_rand_r(UINT32_t* seed) nogil:
 
 
 cdef inline np.ndarray sizet_ptr_to_ndarray(SIZE_t* data, SIZE_t size):
-    """Encapsulate data into a 1D numpy array of intp's."""
+    """Encapsulate data into a 1D numpy array of intp's.
+       You should ensure that the provided data pointer is not freed while the returned array
```

Comment:
It seems like everywhere this is used it is now used with a copy. I propose we just do the copying here. WDYT? @raghavrv?